### ISSUE TITLE
How to use ngrams in search

### ISSUE THREAD

**Issue by uris77** [pacedotdev/firesearch-sdk#5]
TLDR: I'm trying to search by the ngram tokens but that doesn't seem to be working.

I created a document that has a date as a search field. I noticed that the date has some `ngram` tokens. For example, the date `2021-05-11` gets these tokens:

<img width="436" alt="Screen Shot 2021-04-24 at 8 52 39 AM" src="https://user-images.githubusercontent.com/1095741/115962838-6dfc2900-a4da-11eb-88e0-eaf9e54e3e0f.png">

However, I don't get any results back when I search using those tokens.

Is it allowed to search by the `ngram` tokens? If so, is there some configuration we need to tweak?

PS. I'm using the default values for the index:

```
createIndexReq := firesearch.CreateIndexRequest{
		Index: firesearch.Index{
			IndexPath:     fmt.Sprintf("firesearch/indexes/%s", indexName),
			Name:          name,
			Language:      "english",
			KeepStopWords: false,
			CaseSensitive: false,
			NoStem:        false,
		},
	}
```

**Comment by dahernan** [comment]
Hi, ngrams is a internal field, is not exposed and it is used during search you don't need to do anything to use it, the search automatically takes into account that field.

I tried your example and searching for `2021-05` returning the correct results, so not sure what is the problem here.
Can you provide also the insert and the search code?

You can insert the dates like that, however I wouldn't model a date as a searchable field, because there is not much value tokenizing a field that is just a date, until we support a field type `Date`, I would insert dates like this:

```
"fields": [
			{
				"key": "yearOfArrival",
				"value": "2021"
			},{
				"key": "monthOfArrival",
				"value": "05"
			}{
				"key": "dayOfArrival",
				"value": "11"
			}
		]
```

With this data modeling you can filter all entries by year, month, and date, and that is much more powerfull to compose queries

**Comment by uris77** [comment]
Thank you for your response.

Yes, `2021-05` returned results, but it also included all results for `2021`. I thought that since there were tokens generated for `202105`, that searching for that string alone would work.


Thank you for the date structuring suggestion. I like that.

**Comment by dahernan** [comment]
Yeah that is expected, as there is no efficient way right now to know that you are searching for numbers
